packages/server/src/modules/post/mutations/postCreateMutation.ts

```diff
@@ -0,0 +1,40 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql'
+import { mutationWithClientMutationId } from 'graphql-relay'
+
+import { GraphQLContext } from 'src/modules/graphql/types'
+import { PostModel } from '../PostModel'
+import { PostType } from '../PostType'
+
+export const postCreate = mutationWithClientMutationId({
+  name: 'PostCreate',
+  inputFields: {
+    title: { type: new GraphQLNonNull(GraphQLString) },
+    body: { type: new GraphQLNonNull(GraphQLString) }
+  },
+  mutateAndGetPayload: async ({ title, body }, ctx: GraphQLContext) => {
+    if (!ctx?.user) {
+      throw new Error('You are not logged in!')
+    }
```

Comment:
You could add it as a middleware.

Reply:
I will improve this in the next pull request, thank you!